backend/Events/GameStartEvents.js

```diff
@@ -0,0 +1,76 @@
+const MafiaGame = require("../domain/MafiaGame");
+const roles = require("../domain/Enum/Role");
+const GameStartDTO = require("../domain/DTO/response/GameStartDTO");
+const config = require("../config.json");
+
+/**
+ * Event handler of `start-game`
+ * The game has started! Broadcast individual roles to everyone.
+ * @param {any} socket
+ * @param {mafiaGame} mafiaGame
+ */
+function startGame(socket, mafiaGame) {
+  socket.on("start-game", () => {
+    const room = mafiaGame.gameRoomsDict[socket.player.roomID];
+    const players = room.players;
+    const availableRoles = getAvailableRolesToAssign(players.length);
+
+    broadcastRandomRoleToEachPlayer(players, availableRoles);
+  });
+}
+
+function broadcastRandomRoleToEachPlayer(players, availableRoles) {
+  const playersDeepCopy = JSON.parse(JSON.stringify(players));
+
+  while (playersDeepCopy.length > 0) {
+    // Determine random indices to help allocate a random role to a random player
+    let randomRoleIndex = Math.floor(Math.random() * availableRoles.length);
+    let randomPlayerIndex = Math.floor(Math.random() * playersDeepCopy.length);
+
+    const role = availableRoles[randomRoleIndex];
+    const player = playersDeepCopy[randomPlayerIndex];
+
+    player.role = role;
+    socket.broadcast.to(player.socketID).emit("game-start", new GameStartDTO(role));
+
+    // Delete the player and the role which was just allocated
+    availableRoles.splice(randomRoleIndex, 1);
+    playersDeepCopy.splice(randomPlayerIndex, 1);
+  }
+}
+
+function getAvailableRolesToAssign(numOfPlayers) {
+  const roleLogic = config.role_distribution_logic;
```

Comment:
@parfei @MKTPR what do you guys reckon? @LindaBot which domain class would it be added to though?

Reply:
Ah yes, I definitely agree with moving some of the logic out of the event handler. Not sure if it should be moved to the domain model, but it can be for now! :)

Reply:
For now its fine where it is. First priority is to get everyone's work up as quickly as possible for integration. Can deal with these architecture problems when the whole code is up. Let's not delay pull requests any longer.

Reply:
The branch kept getting outdated and had a lot of merge conflicts so I rebased and merged it in to avoid stalling any future work. Will add the tests and move the logic in a new PR